ecosystem/sep-xxxx.md

```diff
@@ -0,0 +1,182 @@
+
+
+## Abstract
+We present an efficient encoding mechanism to store arbitrary binary data on the Stellar ledger via the `ManageData` operation. It will concatenate a MIME-type alongside the aforementioned encoded-text as keys and raw binary data as values.
+
+
+## Specification
+First, we will discuss a common NFT issuance model and its ownership implications. This is not a required model, but sets the stage for why a standardized encoding of digital assets is necessary.
+
+### NFT Issuance
+Given an arbitrary digital asset, here's how issuance typically works:
+
+  1. An account is created to represent the digital asset on the ledger
+  2. The account stores the entire digital asset within its managed data entries.
+  3. The account issues a Stellar asset representing ownership over the digital asset we just stored.
+  4. Any account holding that Stellar asset lays claim to the digital asset.
+
+Notice that ownership of the token directly conveys ownership of a unit of the asset, entirely within the Stellar ledger. Other scenarios such as locking down the issuing account to avoid supply increases, ownership transfers via payments, etc. are built on top of this basic model and are thus extraneous to this SEP.
+
+Our focus is Step 2: storing the asset within the account.
+
+### Encoding Binary Data
+We begin by describing how binary data should be represented in a `ManageData` entry.
+
+Since the _key_ is restricted specifically to printable characters (ref: Stellar Core's [`isString32Valid()`](https://github.com/stellar/stellar-core/blob/b031954458df3bb31d8d98f136c6fee40523a10d/src/util/types.cpp#L89-L101) and C's [`isprint()`](https://en.cppreference.com/w/cpp/string/byte/isprint)), this gives us an encoding space of 94 characters or ~6.5 bits per character (a ~25% reduction in the space). From some cursory research, the [BasE91] encoding scheme closest to this theoretical limit and outperforms base64 encoding.
+
+The _value_ can be any binary data; thus, no encoding is necessary.
+
+### Representing the Data
+With the encoding described [above](#encoding-binary-data), we can now discuss the actual storage. Again, we'll use the `ManageData` operation to store the binary data. 
+
+The first entry _may_ be a [MIME type], followed by the first chunk of data, separated by a dash character (`-`). The MIME type *must* fit in a single key, and so must be <= 63 characters long.
+
+If the MIME type is excluded, the dash _must_ still be included so that readers can easily distinguish where the type starts and the data begins.
+
+Every subsequent entry is simply the next BasE91-encoded chunk, sized to fit into the 64-length key string, and the next 64-byte binary chunk.
+
+For example,
+
+| Key | Value |
+|:----|:-----:|
+| text/plain-'ZP^gp@... | .. 64 bytes .. |
+| Io1Tc%ZEYkE#^IR`0e... | .. 64 bytes .. |
+| %yO){B&mA#_1:W8{lT... | .. 64 bytes .. |
+| Cvnu8jZBP>1]TX2$g%B'  | <empty> |
```

Comment:
Unfortunately an empty value cannot be set for a data entry. Doing so has the same affect as deleting the data entry.

Reply:
Apparently I read "(optional)" in [the docs](https://developers.stellar.org/docs/start/list-of-operations/#manage-data) and ignored the rest entirely. Perhaps a `-` (which isn't in the BasE91 alphabet) makes sense as an `<eof>` of sorts.

Reply:
Addressed in a3842352: I propose using `-` for the key and the final chunk in the value.

Reply:
You actually can submit an empty Buffer as a value just not a `null` or entry non existent value.